### DISCUSSION TITLE
Integration with oxidized

### DISCUSSION THREAD

**Discussion by ghost** [Kircheneer/netlint#7] [Ideas]
I can imagine this running somewhere in the background. 

Could be nice if it's triggered by t.ex. an config change in oxidized backup.
The changed config would be linted and output sent to syslog where it can be picked-up and send to irc/slack/SIEM whatever

**Comment by Kircheneer** [comment]
Thanks for submitting this. I have never used oxidized - can you elaborate a little as to what you except from this tool to make this integration possible?

**Comment by ghost** [comment]
[oxidized ](https://github.com/ytti/oxidized) is a backup tool that runs on a server and logs in periodically on (network) devices.
It downloads it's config, and if something has changed since last login, it will save that new config besides the previous ones.
Therefore it allows you to see the changes between any copies of a config it has stored.

**Comment by Kircheneer** [comment]
Looking at the oxidized documentation it looks like this might already be possible with [hooks](https://github.com/ytti/oxidized/blob/master/docs/Hooks.md#hook-type-exec). Can you verify this?

**Comment by ghost** [comment]
just started looking at that for a siem deployment. (might take a while due to other tasks at hand) but i keep you posted

**Comment by ghost** [comment]
well, *that* was bloody simple ;)
The hooks perfect easy to initiate the linting 

ps: maybe slack, irc or the likes are better to chat/exchange info

**Comment by Kircheneer** [comment]
Do you care to show me how exactly you did this? I would like to throw it into the documentation somewhere.

**Reply by ghost** [reply]
<deleted>
--while running it dawned on me ... the last part can be done a lot easier, i re-write it after dinner--

**Reply by Kircheneer** [reply]
If you have the time, feel free to contribute some configuration/script to a `contrib` folder and write a little bit of documentation, I'm sure this is a use case more users would like to explore.

**Comment by Kircheneer** [comment]
Documentation is now online [here](https://netlint.readthedocs.io/en/latest/integrations/oxidized.html) if you want to take a look 👍

**Reply by ghost** [reply]
very neat! good job!
